### PR TITLE
split off refactoring from PR 20691

### PR DESCRIPTION
**Pull request by WalterBright**
The refactoring needs to be done anyway, and reduces the complexity of https://github.com/dlang/dmd/pull/10951

### CODE REVIEW

File: src/dmd/expressionsem.d [hunk 1]
```diff
@@ -12152,14 +12152,15 @@ private bool checkAddressVar(Scope* sc, UnaExp exp, VarDeclaration v)
...
+                if (exp.type.hasPointers() && v.storage_class & STC.scope_ &&
                     !(sc.flags & SCOPE.debug_) && sc.func.setUnsafe())
                 {
                     exp.error("cannot take address of `scope` %s `%s` in `@safe` function `%s`", p, v.toChars(), sc.func.toChars());
                     return false;
                 }
             }
-            else if (!(sc.flags & SCOPE.debug_) && sc.func.setUnsafe())
+            else if (!(sc.flags & SCOPE.debug_) &&
+                     sc.func.setUnsafe())
```
Comment (Geod24) [inline src/dmd/expressionsem.d:12163]: Unrelated change ?

Comment (WalterBright) [reply src/dmd/expressionsem.d:12163]: No. It's reformatting.

Comment (WalterBright) [reply src/dmd/expressionsem.d:12163]: because 20691 will insert a line between them.

Comment (Geod24) [reply src/dmd/expressionsem.d:12163]: It's an unrelated change. It doesn't have merit on its own (while the rest does).

Those kinds of change are hurtful to readability, because when you go back to it months later, you have no idea why such and such was done, as the context is lost. That's why I always ask, please, have changes that *stands on their own* and not within a context.

I'm going to auto-merge anyway, because the best is the enemy of the good, but please keep this in mind when making commit, just as we keep in mind to make single-commit PRs.
